[PREPROCESSOR] Update with automatic MM

Ivanholyy opened the pull request:
Update MM of resolving headers with automatic MM as previous could lead to memory leaks.

Ivanholyy committed:
Update with automatic MM

Frosty515 reviewed (changes requested):
Hello!

I do like most of these changes, they definitely improve this. I do have a few issues/concerns though.
1. I am unsure whether line 362 will actually work. Have you tested the code?
2. I have intentionally not indented the preprocessor directives around the `std::format` check to make them more obvious.

Ivanholyy committed:
Remove explicit std::format and add new includes for PreProcessor.cpp

Ivanholyy commented:
Hello, indeed!

1. Line 362 sould work, I did not build as I am kinda sure these changes won't harm compiler).
2. Fixed I changed it to use implicit std.

Frosty515 reviewed (changes requested):
Hello!

That isn't what I meant. I prefer that `using namespace std;` never gets used. I was talking about the `#ifdef` stuff. I prefer it to not be indented so it stands out.
Adding those additional includes is good though.

Ivanholyy committed:
Update PreProcessor.cpp

Ivanholyy commented:
Changes are applied.

Frosty515 reviewed (approved):
Hello!

This is good. I'll just check it works fine, then I'll merge it.

Ivanholyy commented:
Thanks!
